dags/multipod/legacy.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A DAG to run tests migrated from the legacy XL ML infrastructure"""
+
+import datetime
+from airflow import models
+from dags import composer_env, gcs_bucket, test_owner
+from dags.vm_resource import TpuVersion, Zone, Project, DockerImage, ClusterName
+from dags.multipod.configs import legacy_unit_test, gke_config
+from dags.multipod.configs.common import SetupMode, Platform
+
+# Run once a day at 9 am UTC (1 am PST)
+SCHEDULED_TIME = "0 9 * * *" if composer_env.is_prod_env() else None
+
+
+for test_mode in [SetupMode.NIGHTLY, SetupMode.STABLE]:
+  with models.DAG(
+      dag_id=f"multipod_legacy_xlml_{test_mode.value}",
+      schedule=SCHEDULED_TIME,
+      tags=["multipod_team", "xlml", "legacy", test_mode.value],
+      start_date=datetime.datetime(2024, 1, 10),
+      catchup=False,
+      concurrency=2,
+  ) as dag:
+    docker_image = (
+        DockerImage.MAXTEXT_JAX_NIGHTLY
+        if test_mode == SetupMode.NIGHTLY
+        else DockerImage.MAXTEXT_JAX_STABLE
+    )
+    current_time = datetime.datetime.now()
+    current_date = current_time.strftime("%Y-%m-%d")
```

Comment:
This will be the DAG parsing time, not the test run time. Can you reuse @zpcore's solution in #179?

Reply:
Removed it for now to unblock the migration, only some of the tests will require it, and it's better to tackle the XPK integration in a separate PR.